Resources/public/js/permission.js

```diff
@@ -0,0 +1,44 @@
+jQuery().ready(function () {
+    var label = new Array();
+    var tmp = '';
+    jQuery('.role_table table th').each(function () {
+        tmp = jQuery(this).html().trim();
+        if (tmp != 'MASTER') {
+            label.push(tmp);
+        }
+    });
+    jQuery('.role_table input').on('ifChecked', function(event){
+        var input = jQuery(this);
+        var item = null;
+        var box = null;
+        var slider = null;
+        jQuery.each(label, function (index, value) {
+            item = jQuery('input[value=' + input.val().replace('MASTER', value).replace('OPERATOR', value) + ']');
+            item.prop('checked', input.is(':checked'));
+            item.parent().addClass('checked');
+        });
+    });
+    jQuery('input[value$="MASTER"], input[value$="OPERATOR"]').change(function() {
+        var input = jQuery(this);
+        var item = null;
+        var box = null;
+        var slider = null;
```

Comment:
`item`, `box` & `slider` should not be declared here.
Also, this is javascript. Do not initialize to null unless you actually need it;
```js
var item, box, slider;
```